ezyquant/report.py

```diff
@@ -553,6 +555,279 @@ def to_excel(self, path: str):
+        match_dates : bool, optional
+            Match the dates of strategy and benchmark returns, by default False.
+
+        Returns
+        -------
+        None
+            The function generates performance metrics and visualizations based on the provided parameters.
+        """
+        return qs.reports.basic(
+            returns=self._select_port(
+                result=self.summary_df, with_dividend=with_dividend
+            ),
+            benchmark=benchmark,
+            rf=rf,
+            grayscale=grayscale,
+            figsize=figsize,
+            display=display,
+            compounded=compounded,
+            periods_per_year=periods_per_year,
+            match_dates=match_dates,
+        )
+
+    def to_full(
+        self,
+        with_dividend=True,
+        benchmark: Optional[pd.DataFrame] = None,
+        rf=0.0,
+        grayscale=False,
+        figsize=(8, 5),
+        display=True,
+        compounded=True,
+        periods_per_year=252,
+        match_dates=False,
+    ):
+        """Calculate performance metrics and generate visualizations for a given strategy.
+
+        Parameters
+        ----------
+        with_dividend : bool, optional
+            Include dividend values in the portfolio values, by default True.
+        benchmark : Optional[pd.DataFrame], optional
+            Time-series of benchmark returns, by default is None.
+        rf : float, optional
+            Risk-free rate of return, by default is 0.
+        grayscale : bool, optional
+            Generate grayscale visualizations, by default False.
+        figsize : tuple, optional
+            Size of the figure for visualizations, by default (8, 5).
+        display : bool, optional
+            Display the performance metrics, by default True.
+        compounded : bool, optional
+            Calculate compounded returns for performance metrics, by default True.
+        periods_per_year : int, optional
+            Number of periods per year for calculating annualized metrics, by default 252.
+        match_dates : bool, optional
+            Match the dates of strategy and benchmark returns, by default False.
+
+        Returns
+        -------
+        None
+            The function generates performance metrics and visualizations based on the provided parameters.
+        """
+
+        return qs.reports.full(
+            returns=self._select_port(
+                result=self.summary_df, with_dividend=with_dividend
+            ),
+            benchmark=benchmark,
+            rf=rf,
+            grayscale=grayscale,
+            figsize=figsize,
+            display=display,
+            compounded=compounded,
+            periods_per_year=periods_per_year,
+            match_dates=match_dates,
+        )
+
+    """
+    Pipeline To Quantstats Reports
+    """
+
+    @staticmethod
+    def _select_port(result: pd.DataFrame, with_dividend=True) -> pd.Series:
```

Comment:
1. ให้ใช้ `self._nav_df` เพื่อให้มี `self._initial_capital` ในวันแรก
2. rename method (maybe _get_nav_series)

```suggestion
    def _get_nav_series(self, with_dividend:bool=True) -> pd.Series:
        return self._nav_df["portfolio_with_dividend" if with_dividend else "portfolio"]
```